packages/webview_flutter/webview_flutter_wkwebview/test/src/web_kit_webview_widget_test.dart

```diff
@@ -206,6 +214,279 @@ void main() {
+            method: WebViewRequestMethod.get,
+            headers: <String, String>{'a': 'header'},
+          ));
+
+          final NSUrlRequest request =
+              verify(mockWebView.loadRequest(captureAny)).captured.single
+                  as NSUrlRequest;
+          expect(request.url, 'https://www.google.com');
+          expect(request.allHttpHeaderFields, <String, String>{'a': 'header'});
+          expect(request.httpMethod, 'get');
+        });
+
+        testWidgets('POST without body', (WidgetTester tester) async {
+          await buildWidget(tester);
+
+          await testController.loadRequest(WebViewRequest(
+            uri: Uri.parse('https://www.google.com'),
+            method: WebViewRequestMethod.post,
+          ));
+
+          final NSUrlRequest request =
+              verify(mockWebView.loadRequest(captureAny)).captured.single
+                  as NSUrlRequest;
+          expect(request.url, 'https://www.google.com');
+          expect(request.httpMethod, 'post');
+        });
+
+        testWidgets('POST with body', (WidgetTester tester) async {
+          await buildWidget(tester);
+
+          await testController.loadRequest(WebViewRequest(
+              uri: Uri.parse('https://www.google.com'),
+              method: WebViewRequestMethod.post,
+              body: Uint8List.fromList('Test Body'.codeUnits)));
+
+          final NSUrlRequest request =
+              verify(mockWebView.loadRequest(captureAny)).captured.single
+                  as NSUrlRequest;
+          expect(request.url, 'https://www.google.com');
+          expect(request.httpMethod, 'post');
+          expect(
+            request.httpBody,
+            Uint8List.fromList('Test Body'.codeUnits),
+          );
+        });
+      });
+
+      testWidgets('canGoBack', (WidgetTester tester) async {
+        await buildWidget(tester);
+
+        when(mockWebView.canGoBack).thenAnswer(
+          (_) => Future<bool>.value(false),
+        );
+        expect(testController.canGoBack(), completion(false));
+      });
+
+      testWidgets('canGoForward', (WidgetTester tester) async {
+        await buildWidget(tester);
+
+        when(mockWebView.canGoForward).thenAnswer(
+          (_) => Future<bool>.value(true),
+        );
+        expect(testController.canGoForward(), completion(true));
+      });
+
+      testWidgets('goBack', (WidgetTester tester) async {
+        await buildWidget(tester);
+
+        await testController.goBack();
+        verify(mockWebView.goBack());
+      });
+
+      testWidgets('goForward', (WidgetTester tester) async {
+        await buildWidget(tester);
+
+        await testController.goForward();
+        verify(mockWebView.goForward());
+      });
+
+      testWidgets('reload', (WidgetTester tester) async {
+        await buildWidget(tester);
+
+        await testController.reload();
+        verify(mockWebView.reload());
+      });
+
+      testWidgets('evaluateJavascript', (WidgetTester tester) async {
```

Comment:
To cover the issue mentioned in my other comment: we should have two tests for `evaulateJavascript`, one where the mock webview returns something and one where it returns null.

Reply:
Done. From testing it looks like a `nil` return value for `evaluateJavaScript` would return the String `(null)`. This implementation returns `null`. This is because `toString` is handled differently with Objective-C vs Dart. 

In this implementation, `WKWebView.evaluateJavaScript` returns a `Future<Object?>`. But the current implementation turns the return value to a String before it is returned: https://github.com/flutter/plugins/blame/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FlutterWebView.m#L384. 

Should we make these align? I was considering making `WKWebView.evaluateJavaScript` return `Future<String>`. Then change it to `Future<Object?>` in `webview_flutter` 4.0.0

Reply:
> Done. From testing it looks like a `nil` return value for `evaluateJavaScript` would return the String `(null)`. This implementation returns `null`. [...] Should we make these align?

Yes, good catch. We should have it (just the legacy `evaluateJavascript` implementation) translate `null` to `'(null)'` for consistent behavior, since people may have been relying on the exact string that the ObjC implementation would produce.

> I was considering making `WKWebView.evaluateJavaScript` return `Future<String>`. Then change it to `Future<Object?>` in `webview_flutter` 4.0.0

`WKWebView.evaluateJavaScript` should always return `Future<Object?>`, because that's what the wrapped API returns. `WebKitWebViewPlatformController.evaluateJavascript` is where we should do special handling.